Crypto allowance acceptance fails in previewnet on allowance deletion

### Description

As reported in the [mirror-node Slack channel](https://swirldslabs.slack.com/archives/C03DNJ4UD37/p1692710351551929), the recently added crypto allowance acceptance test is now failing due to the fact the mirror node REST API was recently updated to no longer return deleted crypto allowances.

The current acceptance test invokes the API to verify for the allowance (owner+spender) the `amount` and `amount_granted` fields are both zero.

### Steps to reproduce

 Run the `cryptoAllowance.feature` acceptance test targeting previewnet, and the very last step, verifying via the REST API that the allowance has been deleted fails:
```
Step failed
java.lang.AssertionError: 
Expecting actual not to be empty
	at com.hedera.mirror.test.e2e.acceptance.steps.AccountFeature.verifyMirrorAPIApprovedCryptoAllowanceResponse(AccountFeature.java:237)
	at com.hedera.mirror.test.e2e.acceptance.steps.AccountFeature.verifyCryptoAllowanceDelete(AccountFeature.java:273)
	at ✽.the mirror node REST API should confirm the crypto allowance deletion(file:///Users/user/Development/mirrornode/repos/hedera-mirror-node/hedera-mirror-test/src/test/resources/features/account/cryptoAllowance.feature:19)
```

### Additional context

_No response_

### Hedera network

previewnet

### Version

0.87.1

### Operating system

macOS

--- a/hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/AccountFeature.java
+++ b/hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/AccountFeature.java
@@ -268,9 +268,14 @@ public class AccountFeature extends AbstractFeature {
         setCryptoAllowance(spender, 0);
     }
 
-    @Then("the mirror node REST API should confirm the crypto allowance deletion")
+    @Then("the mirror node REST API should confirm the crypto allowance no longer exists")
     public void verifyCryptoAllowanceDelete() {
-        verifyMirrorAPIApprovedCryptoAllowanceResponse(0L, 0L);
+        verifyMirrorTransactionsResponse(mirrorClient, HttpStatus.OK.value());
+
+        var owner = accountClient.getClient().getOperatorAccountId().toString();
+        var spender = spenderAccountId.getAccountId().toString();
+        var mirrorCryptoAllowanceResponse = mirrorClient.getAccountCryptoAllowanceBySpender(owner, spender);
+        assertThat(mirrorCryptoAllowanceResponse.getAllowances()).isEmpty();
     }
 
     private void setCryptoAllowance(String accountName, long amount) {

--- a/hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/TokenFeature.java
+++ b/hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/TokenFeature.java
@@ -131,6 +131,25 @@ public class TokenFeature extends AbstractFeature {
         verifyMirrorAPIApprovedTokenAllowanceResponse(tokenId, spenderAccountId, approvedAmount, 0);
     }
 
+    @Then("the mirror node REST API should confirm the approved allowance for {string} and {string} no longer exists")
+    @RetryAsserts
+    public void verifyTokenAllowanceDelete(String tokenName, String accountName) {
+        verifyMirrorTransactionsResponse(mirrorClient, HttpStatus.OK.value());
+
+        var owner = accountClient.getClient().getOperatorAccountId().toString();
+        var spender = accountClient
+                .getAccount(AccountClient.AccountNameEnum.valueOf(accountName))
+                .getAccountId()
+                .toString();
+        var token = tokenClient
+                .getToken(TokenClient.TokenNameEnum.valueOf(tokenName))
+                .tokenId()
+                .toString();
+
+        var mirrorTokenAllowanceResponse = mirrorClient.getAccountTokenAllowanceBySpender(owner, token, spender);
+        assertThat(mirrorTokenAllowanceResponse.getAllowances()).isEmpty();
+    }
+
     @Then("the mirror node REST API should confirm the debit of {long} from {string} allowance of {long} for {string}")
     @RetryAsserts
     public void verifyMirrorAPIApprovedDebitedTokenAllowanceResponse(

--- a/hedera-mirror-test/src/test/resources/features/account/cryptoAllowance.feature
+++ b/hedera-mirror-test/src/test/resources/features/account/cryptoAllowance.feature
@@ -16,7 +16,7 @@ Feature: Account Crypto Allowance Coverage Feature
     When I approve <spender> to transfer up to <approvedAmount> tℏ
     Then the mirror node REST API should confirm the approved <approvedAmount> tℏ crypto allowance
     When I delete the crypto allowance for <spender>
-    Then the mirror node REST API should confirm the crypto allowance deletion
+    Then the mirror node REST API should confirm the crypto allowance no longer exists
     Examples:
       | spender | approvedAmount | recipient | transferAmount |
       | "BOB"   | 100            | "ALICE"   | 1              |

--- a/hedera-mirror-test/src/test/resources/features/token/tokenAllowance.feature
+++ b/hedera-mirror-test/src/test/resources/features/token/tokenAllowance.feature
@@ -14,7 +14,7 @@ Feature: Account Crypto Allowance Coverage Feature
     Given I approve <spender> to transfer up to <approvedAmount> of token <tokenName>
     Then the mirror node REST API should confirm the approved allowance <approvedAmount> of <tokenName> for <spender>
     When I delete the allowance on token <tokenName> for <spender>
-    Then the mirror node REST API should confirm the approved allowance 0 of <tokenName> for <spender>
+    Then the mirror node REST API should confirm the approved allowance for <tokenName> and <spender> no longer exists
     Examples:
       | tokenName  | spender | recipient | approvedAmount | transferAmount |
       | "FUNGIBLE" | "BOB"   | "ALICE"   | 10000          | 100            |

